File: app/src/main/java/com/savvasdalkitsis/gameframe/infra/preferences/RxSharedPreferences.java
package com.savvasdalkitsis.gameframe.infra.preferences;

import android.content.SharedPreferences;

import rx.Observable;

public class RxSharedPreferences {

    private final SharedPreferences sharedPreferences;

    public RxSharedPreferences(SharedPreferences sharedPreferences) {
        this.sharedPreferences = sharedPreferences;
    }

    public Observable<String> getString(String key) {
        return Observable.just(sharedPreferences.getString(key, null));
    }

    public void setString(String key, String value) {
        sharedPreferences.edit()
                .putString(key, value)
                .apply();
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/infra/preferences/RxSharedPreferences.kt
package com.savvasdalkitsis.gameframe.infra.preferences

import android.content.SharedPreferences
import com.savvasdalkitsis.gameframe.infra.kotlin.toMaybe
import io.reactivex.Maybe

class RxSharedPreferences(private val sharedPreferences: SharedPreferences) {

    fun getString(key: String): Maybe<String> = sharedPreferences.getString(key, null).toMaybe()

    fun setString(key: String, value: String) {
        sharedPreferences.edit()
                .putString(key, value)
                .apply()
    }
}
